test/langtools/jdk/javadoc/doclet/testSnippetTag/TestSnippetTag.java

```diff
@@ -0,0 +1,1878 @@
+                                              3 // @start region=foo @end
+                                              """);
+        ClassBuilder classBuilder = new ClassBuilder(tb, "pkg.A")
+                .setModifiers("public", "class");
+        forEachNumbered(snippets, (s, i) -> {
+            classBuilder.addMembers(
+                    MethodBuilder.parse("public void case%s() { }".formatted(i)).setComments(s));
+        });
+        classBuilder.write(srcDir);
+        javadoc("-d", outDir.toString(),
+                "-sourcepath", srcDir.toString(),
+                "pkg");
+        checkExit(Exit.OK);
+        for (int j = 0; j < snippets.size(); j++) {
+            checkOutput("pkg/A.html", true,
+                        """
+                        <span class="element-name">case%s</span>()</div>
+                        <div class="block">
+                        <pre class="snippet">
+                        2</pre>
+                        </div>
+                        """.formatted(j));
+        }
+    }
+
+    @Test
+    public void testComment(Path base) throws Exception {
+        record TestCase(Snippet snippet, String expectedOutput) { }
+        final var testCases = List.of(
+                new TestCase(newSnippetBuilder()
+                                     .body("""
+                                           // // @replace substring="//" replacement="Hello"
+                                           ,
+                                            Snippet!""")
+                                     .build(),
+                             """
+                             Hello
+                             ,
+                              Snippet!"""
+                )
+                ,
+                new TestCase(newSnippetBuilder()
+                                     .body("""
+                                           //             // @replace substring="//" replacement="Hello"
+                                           ,
+                                            Snippet!""")
+                                     .build(),
+                             """
+                             Hello
+                             ,
+                              Snippet!"""
+                )
+                ,
+                new TestCase(newSnippetBuilder()
+                                     .body("""
+                                           // // @replace substring="//" replacement=" Hello"
+                                           ,
+                                            Snippet!""")
+                                     .build(),
+                             """
+                              Hello
+                             ,
+                              Snippet!"""
+                )
+// Uncomment when parser has improved (this would allow to write meta snippets,
+// i.e. snippets showing how to write snippets.
```

Comment:
Just a question: what are the limitations in the parser that are getting in the way here?

Reply:
Both the commented-out code and its description are from a previous version of markup; you can see that the syntax is slightly different from what we have in JEP today. It's good that you asked this question, because this piece is outdated.

Although the markup syntax has changed, the idea for the test remained: there has to be a way to produce a snippet that showcases markup. I will have to implement that idea.